init.lua

```diff
@@ -838,6 +840,23 @@ function PaperWM:switchToSpace(index)
     Spaces.gotoSpace(space)
 end
 
+function PaperWM:incrementSpace(direction)
+    local curr_space_id = Spaces.focusedSpace()
+    local layout = Spaces.allSpaces()
+    local curr_uuid = Screen.mainScreen():getUUID()
+    local curr_space_idx = 0
+    for idx, space_id in ipairs(layout[curr_uuid]) do
+        if curr_space_id == space_id then 
+            curr_space_idx = idx
+        end
+    end
+    if direction == Direction.LEFT then
+        self:switchToSpace((curr_space_idx - 2) % #layout[curr_uuid] + 1)
```

Comment:
don't you need to index the `layout[curr_uuid]` table to get the desired space id?

Reply:
if you guard that direction only equals Direction.LEFT or Direction.RIGHT you could use their -1 and 1 values to do something like:

```
local new_space_idx = ((i - 1 + direction) % #layout[curr_uuid]) + 1
local new_space_id = layout[curr_uuid][new_space_idx]
self:switchToSpace(new_space_id)
```

Reply:
> don't you need to index the `layout[curr_uuid]` table to get the desired space id?

do you mean w.r.t `curr_space_idx`? because that is set from `layout[curr_uuid]` in the for loop:

```suggestion
    for idx, space_id in ipairs(layout[curr_uuid]) do
        if curr_space_id == space_id then 
            curr_space_idx = idx
        end
```
The only other reference to layout in that line is just to get the length i.e. the number of desktops in that uuid table.